src/plugins/data/server/search/types.ts

```diff
@@ -44,68 +49,15 @@ export interface ISearchOptions {
   signal?: AbortSignal;
 }
 
-/**
- * Contains all known strategy type identifiers that will be used to map to
- * request and response shapes. Plugins that wish to add their own custom search
- * strategies should extend this type via:
- *
- * const MY_STRATEGY = 'MY_STRATEGY';
- *
- * declare module 'src/plugins/search/server' {
- *  export interface IRequestTypesMap {
- *    [MY_STRATEGY]: IMySearchRequest;
- *  }
- *
- *  export interface IResponseTypesMap {
- *   [MY_STRATEGY]: IMySearchResponse
- *  }
- * }
- */
-export type TStrategyTypes = typeof ES_SEARCH_STRATEGY | string;
-
-/**
- * The map of search strategy IDs to the corresponding request type definitions.
- */
-export interface IRequestTypesMap {
-  [ES_SEARCH_STRATEGY]: IEsSearchRequest;
-  [key: string]: IKibanaSearchRequest;
-}
-
-/**
- * The map of search strategy IDs to the corresponding response type definitions.
- */
-export interface IResponseTypesMap {
-  [ES_SEARCH_STRATEGY]: IEsSearchResponse;
-  [key: string]: IKibanaSearchResponse;
-}
-
-export type ISearch<T extends TStrategyTypes> = (
-  context: RequestHandlerContext,
-  request: IRequestTypesMap[T],
-  options?: ISearchOptions
-) => Promise<IResponseTypesMap[T]>;
-
-export type ISearchCancel<T extends TStrategyTypes> = (
-  context: RequestHandlerContext,
-  id: string
-) => Promise<void>;
-
 /**
  * Search strategy interface contains a search method that takes in a request and returns a promise
  * that resolves to a response.
  */
-export interface ISearchStrategy<T extends TStrategyTypes> {
-  search: ISearch<T>;
-  cancel?: ISearchCancel<T>;
+export interface ISearchStrategy {
+  search: (
+    context: RequestHandlerContext,
+    request: IEsSearchRequest,
+    options?: ISearchOptions
+  ) => Promise<IEsSearchResponse>;
```

Comment:
I think this is fine for now but we may need to have a chat about how solutions teams/plugin developers may want to eventually return custom responses (other than `IEsSearchResponse`).